v2/streaming-data-generator/src/main/java/com/google/cloud/teleport/v2/templates/StreamingDataGenerator.java

```diff
@@ -97,144 +100,180 @@
-      }
+    /**
+     * The {@link MessageGeneratorFn} class generates {@link PubsubMessage} objects from a supplied
+     * schema and populating the message with fake data. Also supports supplying schema for PubSub
+     * message attributes.
+     *
+     * <p>See <a href="https://github.com/vincentrussell/json-data-generator">json-data-generator</a>
+     * for instructions on how to construct the schema file.
+     */
+    static class MessageGeneratorFn extends DoFn<Long, PubsubMessage> {
+
+        private final String schemaLocation;
+        private String attributesSchemaLocation = null;
+        private String schema;
+        private String attributeSchema;
+
+        // Not initialized inline or constructor because {@link JsonDataGenerator} is not serializable.
+        private transient JsonDataGenerator dataGenerator;
+
+        MessageGeneratorFn(String schemaLocation) {
+            this.schemaLocation = schemaLocation;
+        }
 
-      receiver.output(new PubsubMessage(payload, attributes));
+        MessageGeneratorFn(String schemaLocation, String attributesSchemaLocation) {
+            this.schemaLocation = schemaLocation;
+            this.attributesSchemaLocation = attributesSchemaLocation;
+        }
+
+        @Setup
+        public void setup() throws IOException {
+            dataGenerator = new JsonDataGeneratorImpl();
+
+            Metadata metadata = FileSystems.matchSingleFileSpec(schemaLocation);
+
+            // Copy the schema file into a string which can be used for generation.
+            try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
+                try (ReadableByteChannel readerChannel = FileSystems.open(metadata.resourceId())) {
+                    try (WritableByteChannel writerChannel = Channels.newChannel(byteArrayOutputStream)) {
+                        ByteStreams.copy(readerChannel, writerChannel);
+                    }
+                }
+
+                schema = byteArrayOutputStream.toString();
+            }
+
+            if (attributesSchemaLocation != null) {
+                Metadata metadataAttr = FileSystems.matchSingleFileSpec(attributesSchemaLocation);
+                // Copy the schema file into a string which can be used for generation.
+                try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
+                    try (ReadableByteChannel readerChannel = FileSystems.open(metadataAttr.resourceId())) {
+                        try (WritableByteChannel writerChannel = Channels.newChannel(byteArrayOutputStream)) {
+                            ByteStreams.copy(readerChannel, writerChannel);
+                        }
+                    }
+                    attributeSchema = byteArrayOutputStream.toString();
+                }
+            }
+            //Set Timezone to UTC to generate datetime with UTC
+            TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
```

Comment:
Elaborate on why is this required?  By default DataGenerator generates UTC timestamp values

Reply:
Fixed. I implemented this because I ran a simple test outside of beam/dataflow.